pkg/x509/x509.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package x509
+
+import (
+	"crypto"
+	"crypto/x509"
+	"encoding/asn1"
+	"errors"
+
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+)
+
+var (
+	EKUOID             = asn1.ObjectIdentifier{2, 5, 29, 37}
```

Comment:
nit: add a comment here: this is extended key usage OID?

Reply:
updated